app/src/signup.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "./signup.css";
+
+async function SignupUser(credentials) {
+  return fetch("http://localhost:1000/signup", {
+    method: "POST",
+
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(credentials),
+  }).then((Response) => Response.json());
+}
+
+function Signup() {
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [response_message, setresponse_message] = useState("");
+
+  const [passwordError, setpasswordError] = useState("");
+  const [emailError, setemailError] = useState("");
+  const [usernameError, setusernameError] = useState("");
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    let formvalid = handleValidation();
+    if (formvalid === true) {
+      const response = await SignupUser({
+        username,
+        email,
+        password,
+      });
+      console.log(response);
+      setresponse_message(response["message"]);
+    }
+  };
+
+  const handleValidation = (event) => {
+    let formIsValid = true;
+
+    if (!username.match(/^[a-zA-Z]{5,22}$/)) {
+      formIsValid = false;
+      setusernameError("Invalid username");
+      return false;
+    } else {
+      setusernameError("");
+      formIsValid = true;
+    }
+
+    if (!email.match(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/)) {
+      formIsValid = false;
+      setemailError("Email Not Valid");
+      return false;
+    } else {
+      setemailError("");
+      formIsValid = true;
+    }
+
+    if (!password.match(/^[a-zA-Z]{5,22}$/)) {
+      formIsValid = false;
+      setpasswordError("Invalid password");
```

Comment:
Add details about a valid password in your message that it should be 5 to 22 chars long and should contain only alpha chars.